src/Failsafe.ts

```diff
@@ -30,18 +97,22 @@ export class Failsafe {
 
         return scriptsName.reduce((previous: Promise<ExitCode>, script_name: string) => {
             return previous
-                .then(previous_exit_code => this.runScript(script_name)
+                .then(previous_exit_code => this.runScript(script_name, this.scriptArguments[script_name])
                     .then(current_exit_code => Math.max(previous_exit_code, current_exit_code)));
         }, Promise.resolve(0));
     }
 
-    private runScript(script_name: string): Promise<ExitCode> {
+    private runScript(script_name: string, arguments_: string[] = []): Promise<ExitCode> {
         return new Promise((resolve, reject) => {
-            const npm = process.platform.startsWith('win32')
+            const npm = process.env.npm_execpath ?? (process.platform.startsWith('win32')
                 ? `npm.cmd`
-                : `npm`;
+                : `npm`);
```

Comment:
I've added `process.env.npm_execpath` which points to `yarn` in case you're using yarn (like me)… it will probably also influence other package managers as well, but hopefully in a good way.

Reply:
Nice, good idea

Reply:
just had to change it due to windows failing